server/db/orderItems.js

```diff
@@ -16,17 +16,31 @@ function addToOrder (userId, orderId, conn = connection) {
       if (!orderExists) {
         throw new Error('Order does not exist.')
       }
+    })
+    .then(() => {
       return users.getUser(userId, conn)
         .then(user => {
           if (!user) {
             throw new Error('User does not exist.')
           }
           return conn('order_items')
-            .insert({
-              user_id: user.userId,
-              user_name: user.name,
-              order_text: user.orderText,
-              order_id: orderId
+            .where({
+              'user_id': userId,
+              'order_id': orderId
+            })
+            .select()
+            .first()
+            .then(existingUser => {
+              if (existingUser) {
+                throw new Error('User already exists in order')
```

Comment:
We should never throw errors on the server. I think we can use `Promise.reject` so the route will know to send a `500`.